### ISSUE TITLE
Move all bash-completion logic to Haskell (from bash script)

### ISSUE THREAD

**Issue by Martinsos** [wasp-lang/wasp#258]
With https://github.com/wasp-lang/wasp/pull/224, we added bash-completion for Wasp CLI. Inspect changes introduced by it to get a good grasp of what makes bash-completion work for Wasp.

Logic that does bash-completion is divided into two basic parts: .sh script that is called by bash when it needs completion, and piece of Haskell logic in Wasp CLI that is then invoked by .sh script.

.sh script is something that user has to install manually and ideally we don't to want them to update it or reinstall it with every new version of Wasp. Therefore, we put as much logic as we could into Haskell code, since that is part of Wasp CLI and is therefore part of normal wasp updates. However, we could do more -> we could make bash script even simpler, and move even more logic to Haskell (to Wasp CLI).

This was already discussed here: https://github.com/wasp-lang/wasp/pull/224#discussion_r631285663 -> so this is what we would like to achieve. One mistake is made in the comment I referenced though: when command is handling completion, env vars provided by bash are `COMP_LINE` and `COMP_POINT` (and some more but these are most important), not `COMP_WORDS` and `COMP_CWORDS`. Read more about it here: https://www.gnu.org/software/bash/manual/html_node/Programmable-Completion.html .

So to summarize:
- We want to make `waspc/data/Cli/bash-completion` file simpler, so it uses command instead of bash function, probably it should be just one line saying `complete -o default -o nospace -C "wasp completion:list".
- We should then update code in `waspc/cli/Command/BashCompletion.hs`, mostly `listCommands` but also some things around it, to inspect env vars COMP_LINE and COMP_POINT and return correct list of commands based on that.

That should be about it -> bash completion for Wasp should work after this as it was working before, while enabling us to easily update Wasp in the future without worrying about bash completion script becoming old (since it just redirects to haskell code).

**Comment by auto-kad** [comment]
I'd love to pick up this ticket if it's not already being worked on!

**Comment by Martinsos** [comment]
Pls go for it @auto-kad !

**Comment by Martinsos** [comment]
@auto-kad un-assigning for now, but if you decide to pick it up again let me know!

**Comment by saatvikshah** [comment]
Hi, I'd like to pick up this task.

**Comment by Martinsos** [comment]
Sure @saatvikshah , go for it!

**Comment by Martinsos** [comment]
Done by #305
